### ISSUE TITLE
Parse date using timezone return different values based on format

### ISSUE THREAD

**Issue by MCFreddie777** [moment/moment-timezone#790]
Parsing date using timezone returns different output if used with and without date format.

Current result:
```javascript
moment.tz("18 Oct 2019, 15:40:42", "Europe/London").valueOf()
1571413242000
moment.tz("18 Oct 2019, 15:40:42", "DD MMM YYYY, hh:mm:ss", "Europe/London").valueOf()
1571409642000
```

Expected result:
```javascript
moment.tz("18 Oct 2019, 15:40:42", "Europe/London").valueOf()
1571409642000
moment.tz("18 Oct 2019, 15:40:42", "DD MMM YYYY, hh:mm:ss", "Europe/London").valueOf()
1571409642000
```

This does not look like expected behaviour. 
Result of converting date based on timezone should not be dependent on date formatting.

**Comment by thiagozampieri** [comment]
I tested it here and the result was the same as expected.
What version of the library were you using for this problem?

![image](https://user-images.githubusercontent.com/18358018/67710380-99ae5980-f99e-11e9-8ece-998555f28457.png)

**Comment by thiagozampieri** [comment]
The problem is that moment-timezone only accepts auto parse for RFC2822 or ISO format.

var moment1 = moment.tz("2019-10-18T15:40:42", "America/New_York").valueOf();
var moment2 = moment.tz("18 Oct 2019, 15:40:42", "DD MMM YYYY, hh:mm:ss", "America/New_York").valueOf();

**Comment by ellenaua** [comment]
This is the same bug as https://github.com/moment/moment-timezone/issues/807 - @todo fix it

**Comment by ichernev** [comment]
This is not an issue, just don't assume your format is handled by moment, because it will silently be passed on to `Date` and all bets are off. There is a reason moment constructor outputs warnings if that happens. We should just introduce new constructors which disable this behavior (passing to Date and hoping for the best).

TLDR -- if you KNOW the format, pass it. If you don't know the format, don't get upset when it's not parsed correctly. Every single country has it's own standards and rituals of writing dates, moment (and Date) can't possibly know what you mean.
